lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -38,34 +39,47 @@ const isEqualValueNodes = (nodes1, nodes2) => {
 		const node1 = nodes1[i];
 		const node2 = nodes2[i];
 
+		// Different types indicate different syntaxes.
 		if (typeof node1 === 'undefined' || typeof node2 === 'undefined' || node1.type !== node2.type) {
 			return false;
 		}
 
+		const node1Name = 'name' in node1 ? String(node1.name) : '';
+		const node2Name = 'name' in node2 ? String(node2.name) : '';
+
+		// Custom properties have unknown value syntaxes but are equal for CSS parsers.
+		if (
+			node1.type === 'Identifier' &&
+			node1Name.startsWith('--') &&
+			node2.type === 'Identifier' &&
+			node2Name.startsWith('--')
+		) {
+			continue;
+		}
```

Comment:
Dashed idents are different from regular idents.
These are always known to be the same syntax.